### PR TITLE
feat: re-worked for vulnerabilities becoming part of core schema

### PR DESCRIPTION
**Pull request by madpah**

### CODE REVIEW

File: cyclonedx/model/vulnerability.py [hunk 1]
```diff
@@ -22,21 +22,105 @@
...
+
+    def __init__(self, name: Optional[str] = None, url: Optional[XsUri] = None) -> None:
+        if not name and not url:
+            raise NoPropertiesProvidedException(
+                f'Either name or url must be provided for a VulnerabilitySource - neither provided'
+            )
+        self._name = Optional[str] = name
+        self._url = Optional[XsUri] = url
+
+    def get_name(self) -> Optional[str]:
```
Comment (jkowalleck) [inline cyclonedx/model/vulnerability.py:59]: why not use pythons `@property` for the getters?

File: cyclonedx/model/__init__.py [hunk 2]
```diff
@@ -580,3 +581,237 @@ def get_content_encoding(self) -> Optional[Encoding]:
...
+    .. note::
+        See the CycloneDX Schema definition: https://cyclonedx.org/docs/1.4/xml/#type_organizationalContact
+    """
+
+    def __init__(self, name: Optional[str] = None, phone: Optional[str] = None, email: Optional[str] = None) -> None:
+        if not name and not phone and not email:
+            raise NoPropertiesProvidedException(
+                'One of name, email or phone must be supplied for an OrganizationalContact - none supplied.'
+            )
+        self.name = name
```
[the remark this answers is not in the document]
Comment (jkowalleck) [reply cyclonedx/model/__init__.py:600]: @madpah i think you could try to get your IDE know the private properties by defining them in the class definition. so you would use the setter in the init, and have typing for the private prperties.

like `_bar` in this example: 
```py
class Foo:

  _bar: int

  def __init__(self, b: int) -> None:
    self.bar = b

  @property
  def bar(self) -> int:
    print("getter")
    return self._bar

  @bar.setter
  def bar(self, b: int):
    print("setter")
    self._bar = b

foo = Foo(23)
print(foo.bar)
```

File: tests/test_component.py [hunk 3]
```diff
@@ -114,37 +114,37 @@ def test_custom_package_url_type(self) -> None:
...
-        self.assertEqual(c.get_name(), 'fixtures/bom_v1.3_setuptools.xml')
-        self.assertEqual(c.get_version(), '0.0.0-16932e52ed1e')
+        self.assertEqual(c.name, 'fixtures/bom_v1.3_setuptools.xml')
+        self.assertEqual(c.version, '0.0.0-16932e52ed1e')
         purl = PackageURL(
             type='generic', name='fixtures/bom_v1.3_setuptools.xml', version='0.0.0-16932e52ed1e'
         )
         self.assertEqual(c.to_package_url(), purl)
-        self.assertEqual(len(c.get_hashes()), 1)
+        self.assertEqual(len(c.hashes), 1)
```
[the remark this answers is not in the document]
Comment (madpah) [reply tests/test_component.py:123]: @sonatype-lift ignore

File: tests/test_model_component.py [hunk 4]
```diff
@@ -47,30 +47,30 @@ def test_external_references(self) -> None:
...
-        self.assertEqual(c.get_name(), 'test-component')
-        self.assertEqual(c.get_version(), '')
-        self.assertEqual(c.get_type(), ComponentType.LIBRARY)
-        self.assertEqual(len(c.get_external_references()), 0)
-        self.assertEqual(len(c.get_hashes()), 0)
+        self.assertEqual(c.name, 'test-component')
+        self.assertIsNone(c.version, None)
+        self.assertEqual(c.type, ComponentType.LIBRARY)
+        self.assertEqual(len(c.external_references), 0)
+        self.assertEqual(len(c.hashes), 0)
```
[the remark this answers is not in the document]
Comment (madpah) [reply tests/test_model_component.py:75]: @sonatype-life ignore

**Pull request merged**